rclcpp/src/rclcpp/node_interfaces/node_base.cpp

```diff
@@ -310,3 +316,51 @@ NodeBase::resolve_topic_or_service_name(
   allocator.deallocate(output_cstr, allocator.state);
   return output;
 }
+
+rclcpp::node_interfaces::map_of_mutexes NodeBase::map_object = map_of_mutexes();
+
+rclcpp::node_interfaces::map_of_mutexes::map_of_mutexes()
+{
+}
```

Comment:
Since this is the default constructor, we don't need to define this.  Just remove this (and the declaration in the class).